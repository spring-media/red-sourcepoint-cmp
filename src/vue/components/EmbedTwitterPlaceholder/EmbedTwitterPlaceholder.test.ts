import { mount } from '@vue/test-utils';
import { EmbedTwitterPlaceholder } from './';

describe('EmbedTwitterPlaceholder', () => {
  it('should render without any errors', () => {
    expect(mount(EmbedTwitterPlaceholder).element).toMatchSnapshot();
  });

  describe('should emit a "requestConsent" event by clicking on', () => {
    it('the "activate consent" button', () => {
      const wrapper = mount(EmbedTwitterPlaceholder);

      wrapper.find('.embed-placeholder__button').trigger('click');

      expect(wrapper.emitted('requestConsent')).toBeTruthy();
    });

    it('the link to the description', () => {
      const wrapper = mount(EmbedTwitterPlaceholder);

      wrapper.find('.embed-placeholder__link-description').trigger('click');

      expect(wrapper.emitted('requestConsent')).toBeTruthy();
    });

    it('the link to the vendor list', () => {
      const wrapper = mount(EmbedTwitterPlaceholder);

      wrapper.find('.embed-placeholder__link-vendor-list').trigger('click');

      expect(wrapper.emitted('requestConsent')).toBeTruthy();
    });
  });
});
