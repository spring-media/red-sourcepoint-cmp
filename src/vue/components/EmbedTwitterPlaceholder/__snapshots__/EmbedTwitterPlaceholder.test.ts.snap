// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedTwitterPlaceholder should render without any errors 1`] = `
<div
  class="embed-placeholder__container"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="84"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="16"
        cy="16"
        fill="#E9ECEF"
        r="16"
      />
       
      <rect
        fill="#E9ECEF"
        height="10"
        width="60"
        x="40"
        y="7"
      />
       
      <rect
        fill="#E9ECEF"
        height="6"
        width="40"
        x="40"
        y="19"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="100%"
        y="44"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="100%"
        y="60"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="75%"
        y="76"
      />
       
      <path
        d="M19.6 1.8456C18.8952 2.1616 18.1344 2.3704 17.3392 2.464C18.1536 1.9768 18.7752 1.204 19.072 0.284C18.3112 0.7336 17.4664 1.0656 16.5704 1.24C15.8504 0.476 14.8272 0 13.6952 0C11.52 0 9.75762 1.7624 9.75762 3.936C9.75762 4.2448 9.79202 4.5448 9.85922 4.8328C6.58562 4.6672 3.68402 3.1008 1.74242 0.716C1.40482 1.2976 1.20962 1.9752 1.20962 2.696C1.20962 4.0616 1.90322 5.268 2.96082 5.9736C2.31522 5.956 1.70882 5.7744 1.17682 5.4824C1.17682 5.4984 1.17682 5.5168 1.17682 5.532C1.17682 7.4408 2.53362 9.0336 4.33682 9.3936C4.00642 9.4832 3.65762 9.5336 3.29842 9.5336C3.04322 9.5336 2.79922 9.5064 2.55602 9.4616C3.05762 11.024 4.51202 12.1648 6.23442 12.1976C4.88402 13.2552 3.18882 13.8848 1.34082 13.8848C1.02242 13.8848 0.711224 13.8664 0.400024 13.8288C2.14722 14.9504 4.21682 15.5992 6.44082 15.5992C13.6848 15.5992 17.6472 9.596 17.6472 4.3928C17.6472 4.2184 17.644 4.0496 17.6352 3.8808C18.404 3.3304 19.072 2.6368 19.6 1.8456Z"
        fill="#4AA1EC"
        style="transform: translateX(calc(100% - 20px));"
      />
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
    
    An dieser Stelle findest du Inhalte aus Twitter
  
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
    Um mit Inhalten aus Twitter und anderen sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen wir
    deine Zustimmung.
  
  </div>
   
  <button
    class="embed-placeholder__button"
  >
    
      soziale Netzwerke aktivieren
    
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
      Ich bin damit einverstanden, dass mir externe Inhalte aus sozialen Netzwerken angezeigt werden. Damit können
      personenbezogene Daten an Drittanbieter übermittelt werden. Mehr dazu findest du in der
      
    <a
      class="embed-placeholder__text-link embed-placeholder__link-description"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Beschreibung dieses Datenverarbeitungszweck
    </a>
    
      und in den Datenschutzinformationen dieser
      
    <a
      class="embed-placeholder__text-link embed-placeholder__link-vendor-list"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Drittanbieter
    </a>
    .
    
  </div>
</div>
`;
